automerge/src/transaction/manual_transaction.rs

```diff
@@ -20,14 +20,15 @@ use super::{CommitOptions, Transactable, TransactionInner};
 /// intermediate state.
 /// This is consistent with `?` error handling.
 #[derive(Debug)]
-pub struct Transaction<'a> {
+pub struct Transaction<'a, Obs: OpObserver> {
     // this is an option so that we can take it during commit and rollback to prevent it being
     // rolled back during drop.
     pub(crate) inner: Option<TransactionInner>,
     pub(crate) doc: &'a mut Automerge,
+    pub op_observer: Obs,
```

Comment:
This feels like it should be private, or at least crate private, we're likely to change this stuff a lot so I think putting this behind an accessor method as in `AutoCommit` would be sensible.

Also, should this be an `Option<Obs>` for cases where you're uninterested in observing?